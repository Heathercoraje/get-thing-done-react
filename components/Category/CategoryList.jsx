import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';


const Wrapper = styled.div`
	padding-left: 0.6rem;
	color: white;
`;
const CategoryList = ({ categories, ...props }) => (
	<Wrapper className="category-list">
		{categories.map(category => (
    <CategoryItem key={category.id} {...props} {...category} />
		))}
  </Wrapper>
);

CategoryList.propTypes = {
	props: PropTypes.object,
	categories: PropTypes.array,
	category: PropTypes.object,
	id: PropTypes.number
}
const CategoryItem = props => (
	<div className="category-div">
		<li
			onClick={props.click}
			value={props.id}
			data-name={props.name}
			className="category-text"
		>
			{props.name}
  </li>
		<button className="button-delete" onClick={props.delete} value={props.id}>
			&#8854;
  </button>
  </div>
);
CategoryItem.propTypes = {
	id: PropTypes.number,
	name: PropTypes.string,
	click: PropTypes.func,
	delete: PropTypes.func
}

export { CategoryList, CategoryItem };
