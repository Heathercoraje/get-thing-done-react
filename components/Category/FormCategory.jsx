import React, { Component } from 'react';
import HandleClickInput from '../../lib/HandleClickInput';

const FormCategory = ({ add }) => {
	const onClickInput = evt => {
		evt.preventDefault();
		const inputElement = document.getElementById('input-cate');
		HandleClickInput(evt, add, inputElement);
	};
	return (
		<form className="form-cate" onSubmit={onClickInput}>
			<input
				placeholder=" New category"
				id="input-cate"
				style={{ display: 'none' }}
				className="input-gen input-cate"
			/>
			<input type="submit" value="+" className="button-add" />
		</form>
	);
};

export default FormCategory;
