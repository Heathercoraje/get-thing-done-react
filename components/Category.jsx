import React, { Component } from 'react';
import styled from 'styled-components';
import CategoryList from './CategoryList';
import FormCategory from './FormCategory';

const Wrapper = styled.div`
	box-sizing: border-box;
	border-top-left-radius: 10px;
	border-bottom-left-radius: 10px;
	padding: 20px;
	background: #393939;
	float: left;
	width: 35%;
	height: 100%;
	display: flex;
	flex-direction: column;
`;

const Title = <h2 className="Title">Get Things Done</h2>;

const Category = props => (
  <Wrapper>
    {Title}
    <CategoryList
      className="category-list"
      click={props.click}
      categories={props.categories}
      selected={props.selected}
      delete={props.delete}
    />
    <FormCategory add={props.add} />
  </Wrapper>
);

export default Category;
