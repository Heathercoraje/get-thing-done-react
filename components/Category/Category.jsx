import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { CategoryList } from './CategoryList';
import FormCategory from './FormCategory';

const Wrapper = styled.div`
	background: #393939;
	border-top-left-radius: 0.2rem;
	border-bottom-left-radius: 0.2rem;
	padding: 2.8rem 1rem 1.5rem 1.5rem;
	box-sizing: border-box;
	display: flex;
	flex-direction: column;
`;

const Category = props => (
	<Wrapper className="category-container">
		<h3 className="logo">Get Things Done</h3>
		<CategoryList {...props} />
		<FormCategory add={props.add} />
  </Wrapper>
);

Category.propTypes = {
	props: PropTypes.object,
	add: PropTypes.func
}

export default Category;
