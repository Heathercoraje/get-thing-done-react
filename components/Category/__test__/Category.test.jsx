import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Category from '../Category';

test('Category renders correctly', () => {
	const component = shallow(<Category />);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});
