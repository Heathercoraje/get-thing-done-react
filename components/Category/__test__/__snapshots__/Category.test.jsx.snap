// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category renders correctly 1`] = `
<styled.div
  className="category-container"
>
  <h3
    className="logo"
  >
    Get Things Done
  </h3>
  <CategoryList />
  <FormCategory />
</styled.div>
`;
